Maple2.Server.Game/GameServer.cs

```diff
@@ -83,21 +89,16 @@ public GameServer(FieldManager.Factory fieldFactory, PacketRouter<GameSession> r
         return fieldFactory.Get(mapId, instanceId);
     }
 
-    public GameEvent? FindEvent<T>() where T : GameEventInfo {
-        if (eventCache.TryGetValue(typeof(T).Name, out GameEvent? gameEvent)) {
-            return gameEvent;
-        }
-
-        using GameStorage.Request db = gameStorage.Context();
-        gameEvent = db.FindEvent(typeof(T).Name);
-        if (gameEvent != null) {
-            gameEvent.EventInfo.Id = gameEvent.Id;
-            eventCache[typeof(T).Name] = gameEvent;
-        }
+    public GameEvent? FindEvent(GameEventType type) {
+        return eventCache.Values.FirstOrDefault(gameEvent => gameEvent.Metadata.Type == type && gameEvent.IsActive());
```

Comment:
do you ever have multiple events with the same type? I'm thinking you can use the `ConcurrentMultiDictionary` with both `eventId` and `type`

Reply:
Yeah, mostly StringBoard does this. The other events not so much.